library/libobject.ml

```diff
@@ -143,6 +145,7 @@ let declare_named_object_full odecl =
   let odecl =
     let oname = make_oname in
     { object_name = odecl.object_name;
+      object_stage = Summary.Stage.Interp;
```

Comment:
Shouldn't this be 
```suggestion
      object_stage = odecl.object_stage;
```
?
If named objects are not allowed to be synterp then add an assert instead of ignoring wrong declarations.